src/Console/Command/CleanCommand.php

```diff
@@ -3,21 +3,68 @@
 namespace PhpTuf\ComposerStager\Console\Command;
 
 use PhpTuf\ComposerStager\Console\Misc\ExitCode;
+use PhpTuf\ComposerStager\Domain\Cleaner;
 use Symfony\Component\Console\Command\Command;
 use Symfony\Component\Console\Input\InputInterface;
 use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Console\Question\ConfirmationQuestion;
 
 class CleanCommand extends Command
 {
     protected static $defaultName = 'clean';
 
+    /**
+     * @var \PhpTuf\ComposerStager\Domain\Cleaner
+     */
+    private $cleaner;
+
+    public function __construct(Cleaner $cleaner)
+    {
+        parent::__construct(static::$defaultName);
+        $this->cleaner = $cleaner;
+    }
+
     protected function configure(): void
     {
         $this->setDescription('Removes the staging directory');
     }
 
     public function execute(InputInterface $input, OutputInterface $output): int
     {
-        return ExitCode::SUCCESS;
+        /** @var string $stagingDir */
+        $stagingDir = $input->getOption('staging-dir');
+
+        if (!$this->cleaner->directoryExists($stagingDir)) {
+            $output->writeln(sprintf('<error>The staging directory does not exist at "%s"</error>', $stagingDir));
+            return ExitCode::FAILURE;
+        }
+
+        if (!$this->confirm($input, $output)) {
+            return ExitCode::FAILURE;
+        }
+
+        try {
+            $this->cleaner->clean($stagingDir);
+            return ExitCode::SUCCESS;
+
+        // Prevent ugly "explosions" from unhandled exceptions by catching and
+        // formatting absolutely anything.
+        } catch (\Throwable $e) {
+            $output->writeln("<error>{$e->getMessage()}</error>");
+            return ExitCode::FAILURE;
+        }
+    }
+
+    public function confirm(InputInterface $input, OutputInterface $output): bool
+    {
+        /** @var bool $noInteraction */
+        $noInteraction = $input->getOption('no-interaction');
```

Comment:
No, it isn't _functionally_ necessary, but (alas) if you inline it there's no way to write that doxygen to tell your IDE that that method definitely returns a string--because according to its own doxygen, `getOption()` can technically return `string|string[]|bool|null`. I happen to know in this case that it can only return a string because that option is defined as not accepting a value. I want my IDE and static analysis tools to know that, too.